src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/FormFileBindingSourceMetadataProvider.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Metadata
+{
+    public class FormFileBindingSourceMetadataProvider : IBindingMetadataProvider
```

Comment:
Why not also take the `BindingSource` as a parameter and then you don't have to make this class tied to the source you're setting up.